runner/host/dmesg.go

```diff
@@ -41,6 +42,21 @@ func NewDMesgWithContext(ctx context.Context, cfg DMesgConfig) (*DMesg, error) {
 	if ctx == nil {
 		ctx = context.Background()
 	}
+	if runtime.GOOS == "darwin" {
```

Comment:
The `cfg` has an OS field in it, and we should use it here instead `runtime.GOOS` I think. We pass the OS into the host runners to make some testing scenarios a bit easier.